src/ProgressOnderwijsUtils/Data/CascadedDelete.cs

```diff
@@ -195,7 +195,7 @@ delete pk
                     var newDelTable = ParameterizedSql.CreateDynamic("[#del_" + delBatch + "]");
                     var whereClause = !table.QualifiedName.EqualsOrdinalCaseInsensitive(childTable.QualifiedName)
                         ? SQL($"where 1=1")
-                        : SQL($"where {keyColumns.Select(col => SQL($"pk.{col}<>fk.{col}")).ConcatenateSql(SQL($" or "))}");
+                        : SQL($"where {keyColumns.Select(col => SQL($"pk.{col}<>fk.{col} or fk.{col} is null")).ConcatenateSql(SQL($" or "))}");

```

Comment:
tricky tricky - ik had iets als 
```suggestion
                        : SQL($"where {childTable.PrimaryKey.Select(col => SQL($"pk.{col.SqlColumnName()}<>fk.{col.SqlColumnName()})).ConcatenateSql(SQL($" or "))}");
```

intuitiever gevonden, maar deze implementatie heeft het subtiele voordeel dat het zelfs sommige non-same-row-but-same-statement cycles aankan.

(was dat met opzet?)

Reply:
#541